### ISSUE TITLE
Errors piped in the terminal when using the event browser

### ISSUE THREAD

**Issue by fschlueter** [nu-radio/NuRadioMC#496]
When opening a simulation with the eventbrowers and checking the traces I get quite some errors in the terminal. See below. Several of the panels/plots in all 4 tabs are empty, however, I guess this is okay because no reconstruction has been attempted on the simulated event.

```
Dash is running on http://127.0.0.1:8080/

INFO - NuRadioReco.eventbrowser.app - Dash is running on http://127.0.0.1:8080/

INFO:NuRadioReco.eventbrowser.app:Dash is running on http://127.0.0.1:8080/

 * Serving Flask app 'NuRadioReco.eventbrowser.app'
 * Debug mode: off
INFO - NuRadioReco.NuRadioRecoio - initializing NuRadioRecoio with file ['/Users/felix/work/rnog/sims/ilse/nur/event_0.nur']
INFO:NuRadioReco.NuRadioRecoio:initializing NuRadioRecoio with file ['/Users/felix/work/rnog/sims/ilse/nur/event_0.nur']
INFO - NuRadioReco.NuRadioRecoio - initializing NuRadioRecoio with file ['/Users/felix/work/rnog/sims/ilse/nur/event_0.nur']
INFO:NuRadioReco.NuRadioRecoio:initializing NuRadioRecoio with file ['/Users/felix/work/rnog/sims/ilse/nur/event_0.nur']
INFO - NuRadioReco.NuRadioRecoio - initializing NuRadioRecoio with file ['/Users/felix/work/rnog/sims/ilse/nur/event_0.nur']
INFO:NuRadioReco.NuRadioRecoio:initializing NuRadioRecoio with file ['/Users/felix/work/rnog/sims/ilse/nur/event_0.nur']
INFO - NuRadioReco.NuRadioRecoio - initializing NuRadioRecoio with file ['/Users/felix/work/rnog/sims/ilse/nur/event_0.nur']
INFO:NuRadioReco.NuRadioRecoio:initializing NuRadioRecoio with file ['/Users/felix/work/rnog/sims/ilse/nur/event_0.nur']
INFO - NuRadioReco.NuRadioRecoio - ... finished in 0 seconds
INFO:NuRadioReco.NuRadioRecoio:... finished in 0 seconds
INFO - NuRadioReco.NuRadioRecoio - ... finished in 0 seconds
INFO:NuRadioReco.NuRadioRecoio:... finished in 0 seconds
INFO - NuRadioReco.NuRadioRecoio - ... finished in 0 seconds
INFO:NuRadioReco.NuRadioRecoio:... finished in 0 seconds
INFO - NuRadioReco.NuRadioRecoio - ... finished in 0 seconds
INFO:NuRadioReco.NuRadioRecoio:... finished in 0 seconds
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/index.py", line 219, in set_event_number
    event_i = np.where(mask[:,0] * mask[:,1])[0][0]
IndexError: index 0 is out of bounds for axis 0 with size 0
ERROR - NuRadioReco.eventbrowser.app - Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/index.py", line 219, in set_event_number
    event_i = np.where(mask[:,0] * mask[:,1])[0][0]
IndexError: index 0 is out of bounds for axis 0 with size 0
ERROR:NuRadioReco.eventbrowser.app:Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/index.py", line 219, in set_event_number
    event_i = np.where(mask[:,0] * mask[:,1])[0][0]
IndexError: index 0 is out of bounds for axis 0 with size 0
ERROR:BaseStation:Stations particle type has not been set. Please call the module `eventTypeIdentifier.run(event, station, mode='forced', forced_event_type='neutrino'/'cosmic_ray')`. This flag is used to differentiate between signals that originate from air vs. signals that originate from within the ice which is needed to e.g. determine if refraction into the ice needs to be considered.
ERROR:BaseStation:Stations particle type has not been set. Please call the module `eventTypeIdentifier.run(event, station, mode='forced', forced_event_type='neutrino'/'cosmic_ray')`. This flag is used to differentiate between signals that originate from air vs. signals that originate from within the ice which is needed to e.g. determine if refraction into the ice needs to be considered.
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/overview_plots/event_overview.py", line 208, in plot_event_overview
    if nurio.get_detector() is not None:
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'
ERROR - NuRadioReco.eventbrowser.app - Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/overview_plots/event_overview.py", line 208, in plot_event_overview
    if nurio.get_detector() is not None:
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'
ERROR:NuRadioReco.eventbrowser.app:Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/overview_plots/event_overview.py", line 208, in plot_event_overview
    if nurio.get_detector() is not None:
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/simulation_plots/sim_event_overview.py", line 41, in update_sim_event_3d
    det = nurio.get_detector()
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'
ERROR - NuRadioReco.eventbrowser.app - Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/simulation_plots/sim_event_overview.py", line 41, in update_sim_event_3d
    det = nurio.get_detector()
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'
ERROR:NuRadioReco.eventbrowser.app:Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/simulation_plots/sim_event_overview.py", line 41, in update_sim_event_3d
    det = nurio.get_detector()
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'


ERROR:BaseStation:Stations particle type has not been set. Please call the module `eventTypeIdentifier.run(event, station, mode='forced', forced_event_type='neutrino'/'cosmic_ray')`. This flag is used to differentiate between signals that originate from air vs. signals that originate from within the ice which is needed to e.g. determine if refraction into the ice needs to be considered.
ERROR:BaseStation:Stations particle type has not been set. Please call the module `eventTypeIdentifier.run(event, station, mode='forced', forced_event_type='neutrino'/'cosmic_ray')`. This flag is used to differentiate between signals that originate from air vs. signals that originate from within the ice which is needed to e.g. determine if refraction into the ice needs to be considered.
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/overview_plots/event_overview.py", line 208, in plot_event_overview
    if nurio.get_detector() is not None:
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'
ERROR - NuRadioReco.eventbrowser.app - Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/overview_plots/event_overview.py", line 208, in plot_event_overview
    if nurio.get_detector() is not None:
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'
ERROR:NuRadioReco.eventbrowser.app:Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/dash.py", line 1273, in dispatch
    ctx.run(
  File "/Users/felix/software/envs/py310_nuRadio/lib/python3.10/site-packages/dash/_callback.py", line 440, in add_context
    output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
  File "/Users/felix/software/nuRadioMC/NuRadioReco/eventbrowser/apps/overview_plots/event_overview.py", line 208, in plot_event_overview
    if nurio.get_detector() is not None:
  File "/Users/felix/software/nuRadioMC/NuRadioReco/modules/io/NuRadioRecoio.py", line 292, in get_detector
    self.__scan_files()  # Maybe we just forgot to scan the file
TypeError: scan_files_function.<locals>.scan_files_2_2() missing 3 required positional arguments: 'self', 'iF', and 'current_byte'

```

**Comment by fschlueter** [comment]
Okay, this code is super confusing:

in `NuRadioRecoio.py` is defined a "member variable"  `self.__scan_files` which is a function:
```
        self.__scan_files = NuRadioReco.modules.io.event_parser_factory.scan_files_function(self.__file_version, self.__file_version_minor)
```
and also a class method with the same name:
```
    def __scan_files(self):
        current_byte = 12  # skip datafile header
        iF = 0
        ....
```
Is there a reason for it?

**Comment by sjoerd-bouma** [comment]
The io functionality is pretty old, I think you'll need @christophwelling or @cg-laser to explain this. I think it opens the file, checks the file version, and then overwrites `__scan_files` with the appropriate io function from `event_parser_factory`.
I'm not sure why this structure was chosen, but it occasionally leads to the hard to parse error you posted above.

The error is this: `get_detector` is called, the function looks for the detector description (which may or may not be stored in the .nur files), doesn't find it, and as a 'fallback' first tries to make sure the file has actually been read in by calling `__scan_files`. In your case, however, the file *has* already been read in (it just doesn't contain a detector description), so `__scan_files` has been overwritten, and the signature no longer matches, leading to the above error.

**Comment by fschlueter** [comment]
Makes sense. But I do not think this overwriting is necessary. So the solution here is to allow an event/nur file to not have stored a detector or is the simulation I am using just buggy and we do not want to support that?

**Comment by sjoerd-bouma** [comment]
I think not having the detector description should be fine, but `get_detector` should fail more gracefully. With regards to the method being overwritten, I'm also not sure that we want/need to keep the way it's done currently, but there I'd want someone more familiar with the implementation to weigh in first.

**Comment by fschlueter** [comment]
I think we actually fixed this @sjoerd-bouma? I will close this. FFTRO it if necessary
